src/nvim/highlight_group.c

```diff
@@ -747,8 +749,6 @@ void set_hl_group(int id, HlAttrs attrs, Dict(highlight) *dict, int link_id)
     normal_sp = g->sg_rgb_sp;
     ui_default_colors_set();
   } else {
-    g->sg_attr = hl_get_syn_attr(0, id, attrs);
-
     // a cursor style uses this syn_id, make sure its attribute is updated.
```

Comment:
Potential for Normal to ever need to hit this check? Should the else be completely discarded?